### ISSUE TITLE
Cannot read property 'value' of undefined with typescript-react-query enabled

### ISSUE THREAD

**Issue by iamhaaamed** [dotansimha/graphql-code-generator#7229]
**Describe the bug**
I'm just trying to integrate graphql-code-generator with typescript-react-query plugin into my project and having this error:
```
 ✔ Parse configuration
  ❯ Generate outputs
    ❯ Generate src/generated/graphql.tsx
      ✔ Load GraphQL schemas
      ✔ Load GraphQL documents
      ✖ Generate
        → Cannot read property 'value' of undefined

 Found 1 error

  ✖ src/generated/graphql.tsx
    TypeError: Cannot read property 'value' of undefined
        at generateQueryKey (/Users/user/Projects/DrivesAndDetours/node_modules/@graphq
l-codegen/typescript-react-query/index.js:24:31)
```

**To Reproduce**
Create a .graphql file, with a query.
Enable typescript-react-query codegen plugin.
Run yarn generate.

My `codegen.yml` config file:
```yml
overwrite: true
schema: 'schema.graphql'
documents: 'src/graphql'
generates:
  src/generated/graphql.tsx:
    plugins:
      - 'typescript'
      - 'typescript-operations'
      - 'typescript-react-query'
```

**Expected behavior**
should generate graphql.ts

**Comment by geoffwa** [comment]
I had a very similar issue with the following stacktrace:
```
 Found 1 error

  ✖ app/javascript/graphql/types.tsx
    TypeError: Cannot read property 'value' of undefined
        at generateQueryKey (~/application/node_modules/@graphql-codegen/typescript-react-query/index.js:29:53)
        at FetchFetcher.generateQueryHook (~/application/node_modules/@graphql-codegen/typescript-react-query/index.js:213:9)
        at ReactQueryVisitor.buildOperation (~/application/node_modules/@graphql-codegen/typescript-react-query/index.js:520:38)
        at ReactQueryVisitor.OperationDefinition (~/application/node_modules/@graphql-codegen/visitor-plugin-common/index.js:2542:33)
        at Object.visit (~/application/node_modules/graphql/language/visitor.js:197:21)
        at Object.oldVisit (~/application/node_modules/@graphql-codegen/plugin-helpers/index.js:517:20)
        at Object.plugin (~/application/node_modules/@graphql-codegen/typescript-react-query/index.js:571:41)
        at executePlugin (~/application/node_modules/@graphql-codegen/core/index.js:38:35)
        at async ~/application/node_modules/@graphql-codegen/core/index.js:167:24
        at async Promise.all (index 2)
```

This was down to me not having a name for a query operation that I was trying to make, I fixed by by changing
```
import gql from "graphql-tag";

export const thingieQuery = gql`
{
  thingies {
        id
        createdAt
  }
}`;
```

to:

```
import gql from "graphql-tag";

export const thingieQuery = gql`
query fetchThingies {
  thingies {
        id
        createdAt
  }
}`;
```

**Comment by charlypoly** [comment]
Hi @geoffwa, @iamhaaamed,

You are right, the `typescript-react-query` does not support anonymous operations (it should).
This is definitely a bug 🐞 

We'll plan to do a fix.

In the meantime, the solution is to use named operations.

Thank you!

**Comment by NovqiGarrix** [comment]
**_I also had a similar issue with the following stacktrace:_**
√ Parse configuration
  > Generate outputs
    > Generate ./src/graphql.ts
      √ Load GraphQL schemas
      √ Load GraphQL documents
      × Generate
        → Cannot read properties of undefined (reading 'value')


 Found 1 error

  ✖ ./src/graphql.ts
    TypeError: Cannot read properties of undefined (reading 'value')


**I solved it by change my GraphQL Documents**

_FROM THIS ONE_:

```graphql
query {
  me {
    data {
      id
      location
      isVerified
      accountType
      role
      profilePicture
      name
      email
      createdAt
      updatedAt
    }
    error
  }
}
```

_TO:_ 

```graphql
query GetMe {
  me {
    data {
      id
      location
      isVerified
      accountType
      role
      profilePicture
      name
      email
      createdAt
      updatedAt
    }
    error
  }
}
```

_In conclusion, you have to give the query or mutation name on all of your .graphql files, so that the typescript-react-query plugin can create queries, and mutations based on that name_

```
  √ Parse configuration
  √ Generate outputs
Done in 1.16s.
```

**Comment by hassanrbh** [comment]
> **_I also had a similar issue with the following stacktrace:_** √ Parse configuration
> 
> > Generate outputs
> > > Generate ./src/graphql.ts
> > √ Load GraphQL schemas
> > √ Load GraphQL documents
> > × Generate
> > → Cannot read properties of undefined (reading 'value')
> 
> Found 1 error
> 
> ✖ ./src/graphql.ts TypeError: Cannot read properties of undefined (reading 'value')
> 
> **I solved it by change my GraphQL Documents**
> 
> _FROM THIS ONE_:
> 
> ```graphql
> query {
>   me {
>     data {
>       id
>       location
>       isVerified
>       accountType
>       role
>       profilePicture
>       name
>       email
>       createdAt
>       updatedAt
>     }
>     error
>   }
> }
> ```
> 
> _TO:_
> 
> ```graphql
> query GetMe {
>   me {
>     data {
>       id
>       location
>       isVerified
>       accountType
>       role
>       profilePicture
>       name
>       email
>       createdAt
>       updatedAt
>     }
>     error
>   }
> }
> ```
> 
> _In conclusion, you have to give the query or mutation name on all of your .graphql files, so that the typescript-react-query plugin can create queries, and mutations based on that name_
> 
> ```
>   √ Parse configuration
>   √ Generate outputs
> Done in 1.16s.
> ```

soo mush thanks
